samples/extensions/khr/externalmemory/main.c

```diff
@@ -0,0 +1,714 @@
+                 clbufy);
+    OCLERROR_RET(clSetKernelArg(saxpy, 2, sizeof(cl_mem), &cl_buf_y), error,
+                 clbufy);
+
+    // Acquire OpenCL memory objects created from Vulkan external memory
+    // handles.
+    cl_mem cl_mem_objects[] = { cl_buf_x, cl_buf_y };
+    clEnqueueAcquireExternalMemObjectsKHR_fn
+        clEnqueueAcquireExternalMemObjects =
+            (clEnqueueAcquireExternalMemObjectsKHR_fn)
+                clGetExtensionFunctionAddressForPlatform(
+                    cl_platform, "clEnqueueAcquireExternalMemObjectsKHR");
+    clEnqueueAcquireExternalMemObjects(queue, 2, cl_mem_objects, 0, NULL, NULL);
+
+    // Launch kernel.
+    if (diag_opts.verbose)
+    {
+        printf("done.\nExecuting on device... ");
+        fflush(stdout);
+    }
+
+    cl_event kernel_run;
+    GET_CURRENT_TIMER(dev_start)
+    OCLERROR_RET(clEnqueueNDRangeKernel(queue, saxpy, 1, NULL, &length, &wgs, 0,
+                                        NULL, &kernel_run),
+                 error, clbufy);
+    OCLERROR_RET(clWaitForEvents(1, &kernel_run), error, clbufy);
+    GET_CURRENT_TIMER(dev_end)
+
+    cl_ulong dev_time;
+    TIMER_DIFFERENCE(dev_time, dev_start, dev_end)
+
+    // Release OpenCL memory objects created from Vulkan external memory
+    // handles.
+    clEnqueueReleaseExternalMemObjectsKHR_fn
+        clEnqueueReleaseExternalMemObjects =
+            (clEnqueueReleaseExternalMemObjectsKHR_fn)
+                clGetExtensionFunctionAddressForPlatform(
+                    cl_platform, "clEnqueueReleaseExternalMemObjectsKHR");
+    clEnqueueReleaseExternalMemObjects(queue, 2, cl_mem_objects, 0, NULL, NULL);
+
+    // Concurrently calculate reference saxpy.
+    if (diag_opts.verbose)
+    {
+        printf("done.\nExecuting on host... ");
+    }
+
+    GET_CURRENT_TIMER(host_start)
+    host_saxpy(arr_x, arr_y, a, length);
+    GET_CURRENT_TIMER(host_end)
+    cl_ulong host_time;
+    TIMER_DIFFERENCE(host_time, host_start, host_end)
+
+    if (diag_opts.verbose)
+    {
+        printf("done.\n");
+    }
+
+    // Fetch results.
+    OCLERROR_RET(clEnqueueReadBuffer(queue, cl_buf_y, CL_BLOCKING, 0,
+                                     sizeof(cl_float) * length, (void*)arr_x, 0,
+                                     NULL, NULL),
+                 error, clbufy);
+
+    // Validate solution.
+    for (size_t i = 0; i < length; ++i)
+        if (arr_y[i] != arr_x[i])
+        {
+            printf("Verification failed! %f != %f at index %zu\n", arr_y[i],
+                   arr_x[i], i);
+            error = CL_INVALID_VALUE;
+        }
+    if (error == CL_SUCCESS)
+    {
+        printf("Verification passed.\n");
+    }
+
+    if (!diag_opts.quiet)
+    {
+        printf("Kernel execution time as seen by host: %llu us.\n",
+               (unsigned long long)(dev_time + 500) / 1000);
+
+        printf("Kernel execution time as measured by device:\n");
+        printf("\t%llu us.\n",
```

Comment:
Super minor: the CPP sample doesn't include an extra newline here, and prints the device time on the same line, similar to the host time.

Reply:
Changed as requested!